### PR TITLE
IBC hooks - Lifecycle completed

### PR DESCRIPTION
**Pull request by nicolaslara**
## What is the purpose of the change

This refactors the IBC Ack and IBC Timeout messages from wasmhooks so that they are sent within an enum.

This means implementors will need to implement both (or actively ignore them).

Note that this PR depends on #3698 and should either be merged after it or into it.

## Brief Changelog

 
  - Wrapped IBC lifecycle messages in "ibc_lifecycle_complete"

## Testing and Verifying

all tests still pass

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes )
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (no)
  - How is the feature or change documented? (updated docs)

### CODE REVIEW

File: tests/ibc-hooks/testutils/contracts/counter/src/msg.rs [hunk 1]
```diff
@@ -34,13 +34,30 @@ pub struct GetTotalFundsResponse {
 }
 
 #[cw_serde]
-pub enum SudoMsg {
-    ReceiveAck {
+#[serde(rename = "ibc_lifecycle_complete")]
+pub enum IBCLifecycleComplete {
```
Comment (ValarDragon) [inline tests/ibc-hooks/testutils/contracts/counter/src/msg.rs:38]: LGTM, is the plan to eventually move this into some other crate that can be imported?

**Comment by ValarDragon** [comment]
IBC lifecycle changes LGTM

File: tests/ibc-hooks/testutils/contracts/counter/src/msg.rs (hunk 1 above)
Comment (nicolaslara) [reply tests/ibc-hooks/testutils/contracts/counter/src/msg.rs:38]: yeah, I think the outpost repo will have `packages/` for this and other reusable structs that we can publish. Thought not sure if it should be in osmosis-std instead

**Comment by nicolaslara** [comment]
Closing this in favour of https://github.com/osmosis-labs/osmosis/pull/3927 and https://github.com/osmosis-labs/osmosis/pull/3698

**Comment by nicolaslara** [comment]
Closing this in favour of https://github.com/osmosis-labs/osmosis/pull/3927 and https://github.com/osmosis-labs/osmosis/pull/3698

**Pull request closed without merging**